src/algebraic_geometry/prime_spectrum/maximal.lean

```diff
@@ -73,4 +74,60 @@ instance : t1_space $ maximal_spectrum R :=
 
 lemma to_prime_spectrum_continuous : continuous $ @to_prime_spectrum R _ := continuous_induced_dom
 
+variables (R) [is_domain R] {K : Type v} [field K] [algebra R K] [is_fraction_ring R K]
+
+/-- An integral domain is equal to the intersection of its localizations at all its maximal ideals
+viewed as subalgebras of its field of fractions. -/
+theorem localization_infi_eq_bot :
+  (⨅ v : maximal_spectrum R, localization.subalgebra.of_field K v.as_ideal.prime_compl $
+    le_non_zero_divisors_of_no_zero_divisors $ not_not_intro v.as_ideal.zero_mem) = ⊥ :=
```

Comment:
I wonder if this should be made Into a lemma `ideal.prime_compl_le_non_zero_divisors` as it is a bit long.
If so, it can also be done in a follow up PR since there are other occasions that should be replaced as well.

Reply:
I think there's really only one other obvious instance where this is used so I'll just change it here.

Reply:
There's also `ring_theory/valuation/valuation_subring.lean` line 214.

Reply:
Hopefully that's the last one.